### PR TITLE
feat(bigquery): Add Job#parent_job_id and Job#script_statistics

### PR DESCRIPTION
**Pull request by quartzmo**
* Add `Job#num_child_jobs`
* Add `Job#parent_job_id`
* Add `parent_job` to `Project#jobs`
* Add `Job#script_statistics`

closes: #4810

### CODE REVIEW

File: google-cloud-bigquery/acceptance/bigquery/advanced_test.rb [hunk 1]
```diff
@@ -109,6 +109,66 @@
...
+    );
+    -- Which names appear as words in Shakespeare's plays?
+    SELECT
+    name AS shakespeare_name
+    FROM UNNEST(top_names) AS name
+    WHERE name IN (
+    SELECT word
+    FROM `bigquery-public-data.samples.shakespeare`
+    );
+    SQL
```
Comment (dazuma) [inline google-cloud-bigquery/acceptance/bigquery/advanced_test.rb]: Nit: This block would be more readable if the content (between `<<~SQL` and the closing `SQL`) were indented two more spaces.

Same for the examples in comments below.

Comment (quartzmo) [reply google-cloud-bigquery/acceptance/bigquery/advanced_test.rb]: good idea, i'll do it.

Comment (quartzmo) [reply google-cloud-bigquery/acceptance/bigquery/advanced_test.rb]: Done.

**Comment by quartzmo** [comment]
@shollyman PTAL when you have time, thank you.

File: google-cloud-bigquery/acceptance/bigquery/advanced_test.rb [hunk 2]
```diff
@@ -109,6 +109,66 @@
...
+    child_jobs[0].script_statistics.evaluation_kind.must_equal "STATEMENT"
+    child_jobs[0].script_statistics.stack_frames.wont_be :nil?
+    child_jobs[0].script_statistics.stack_frames.must_be_kind_of Array
+    child_jobs[0].script_statistics.stack_frames.count.must_equal 1
+    child_jobs[0].script_statistics.stack_frames[0].must_be_kind_of Google::Cloud::Bigquery::Job::ScriptStackFrame
+    child_jobs[0].script_statistics.stack_frames[0].start_line.must_equal 10
+    child_jobs[0].script_statistics.stack_frames[0].start_column.must_equal 1
+    child_jobs[0].script_statistics.stack_frames[0].end_line.must_equal 16
+    child_jobs[0].script_statistics.stack_frames[0].end_column.must_equal 2
+    child_jobs[0].script_statistics.stack_frames[0].text.must_equal "SELECT\nname AS shakespeare_name\nFROM UNNEST(top_names) AS name\nWHERE name IN (\nSELECT word\nFROM `bigquery-public-data.samples.shakespeare`\n)"
```
Comment (shollyman) [inline google-cloud-bigquery/acceptance/bigquery/advanced_test.rb]: I'd consider relaxing this exact text matching qualifier to something like has a nonzero len, as the contents of the stack frame may change slightly over time.

Comment (quartzmo) [reply google-cloud-bigquery/acceptance/bigquery/advanced_test.rb]: done

**Pull request merged**